### DISCUSSION TITLE
change new generic constraints

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/csharplang#6364] [Language Ideas] [answered]
when we have to use generics and `where` keyword currently we can use new()
for empty constructors but we cant use new(string,int,double) structure to use parametrized constructor

**Comment by PathogenDavid** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/769 (and many many others: https://github.com/dotnet/csharplang/issues/6095 https://github.com/dotnet/csharplang/issues/4723 https://github.com/dotnet/csharplang/issues/2600 https://github.com/dotnet/csharplang/issues/1974 https://github.com/dotnet/csharplang/issues/1633 https://github.com/dotnet/csharplang/issues/1330 https://github.com/dotnet/csharplang/issues/1574 https://github.com/dotnet/csharplang/issues/1626)

Also related: https://github.com/dotnet/csharplang/issues/110

If you control the types involved, consider using a factory method provided as [a static abstract interface method](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/static-virtual-interface-members).
